doc/SEMANTICS.md

```diff
@@ -88,7 +88,11 @@ Renaming files and directories (`rename`, `renameat`) is not currently supported
 
 File deletion (`unlink`) is not currently supported, but will be [in the future](https://github.com/awslabs/mountpoint-s3/issues/78).
 
-Empty directory removal (`rmdir`) is not supported.
+Empty directory removal (`rmdir`) is not currently supported, but will be [in the future](https://github.com/awslabs/mountpoint-s3/issues/194). The following semantics are proposed:
+
+* `rmdir` will be a local only operation. It will delete a local directory only if it is empty.
+* If the directory marker (i.e. zero-byte objects with `<directory-name>/` key) is present remotely on the S3 bucket, `rmdir` should fail even if it is empty locally.
+* This means if the directory is empty and only present locally, it should be deleted immediately. Any future read or writing to directory( creating a file/subdirectory) should fail.
```

Comment:
I would avoid repeating "if it is empty" too many times - it should be clear also from the paragraph above. Also, maybe make it clearer what we mean by "local" directory. For example something like this: 

```suggestion
* `rmdir` will only delete (empty) local directories created by `mkdir`.
* `rmdir` will fail on directories backed on S3 by a directory marker (i.e. an object with `<directory-name>/` key).
* On success, the directory will be deleted immediately. Subsequent reads or writes to the directory (e.g. creating a file or subdirectory) will fail.
```

Reply:
+1.

I think there's still a bit of ambiguity around what happens if all of the files were deleted remotely, and you try to remove the directory. What happens? Does it disappear anyway when we do a `readdir` because we see there's no children? Or we do a `lookup` on the directory, and see that its not there?